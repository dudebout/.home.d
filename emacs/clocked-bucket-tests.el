;;; clocked-bucket-tests.el --- Tests for clocked-bucket.el

;;; Commentary:

;;; Code:

(require 'ert)
(require 'clocked-bucket)

;; temporary
(require 'home.d-org)

(ert-deftest tautology ()
  (should t)
  (should-not nil))

(ert-deftest my-test ()
  (find-file "./clocked-bucket-tests.org")
  (clocked-bucket-get-clocked-tasks-if (lambda () (home.d/has-property "bucket" "a")))
  (clocked-bucket-get-clocked-tasks-if (lambda () (home.d/has-property "bucket" "b")))
  (clocked-bucket-get-clocked-tasks-in-buckets '((lambda () (home.d/has-property "bucket" "a"))
                                                 (lambda () (home.d/has-property "bucket" "b")))))

(ert-deftest my-test2 ()
  (find-file "./clocked-bucket-tests.org")
  (let ((result "category I\n  context 1\n    task A\n    task B\n  context 2\n    task A\n")
        (tasks (mapcar #'car (clocked-bucket-get-clocked-tasks-if (lambda () (home.d/has-property "bucket" "a"))))))
    (should (equal result (clocked-bucket-display-task-trees (clocked-bucket-compute-task-trees tasks))))))

(ert-deftest display-test ()
  (let ((t1 (task-create :category "category 1"
                         :context "context 1"
                         :name "task 1")))
    (should (equal "category 1\n  context 1\n    task 1" (clocked-bucket-display-tasks (list t1))))))

(ert-deftest task-trees-test ()
  (let* ((t1 (task-create :category "category 1"
                          :context "context 1"
                          :name "task 1"))
         (t2 (task-create :category "category 1"
                          :context "context 2"
                          :name "task 1"))
         (t3 (task-create :category "category 1"
                          :context "context 1"
                          :name "task 2"))
         (l (list t1 t2 t3))
         (result "category 1\n  context 1\n    task 1\n    task 2\n  context 2\n    task 1\n"))
    (should (equal result (clocked-bucket-display-task-trees (clocked-bucket-compute-task-trees l))))))

;;; clocked-bucket-tests.el ends here
