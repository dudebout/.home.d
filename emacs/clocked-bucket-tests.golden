* Allocations
+ Allocation A :: 40.00%
+ Allocation B :: 60.00%
* bucket a (10% A + 90% B): 66%
|----------------+--------+--------+---------------|
| category I     | 50.00% |        |               |
|     context 1  |        | 33.33% |               |
|         task A |        |        | 16.67% (3:00) |
|         task B |        |        | 16.67% (3:00) |
|     context 2  |        | 16.67% |               |
|         task A |        |        | 16.67% (3:00) |
| category II    | 16.67% |        |               |
|     context 1  |        | 16.67% |               |
|         task B |        |        | 16.67% (1:00) |
|----------------+--------+--------+---------------|
* bucket b (A): 33%
|----------------+--------+--------+---------------|
| category I     | 16.67% |        |               |
|     context 2  |        | 16.67% |               |
|         task B |        |        | 16.67% (1:00) |
| category II    | 16.67% |        |               |
|     context 1  |        | 16.67% |               |
|         task A |        |        | 16.67% (1:00) |
|----------------+--------+--------+---------------|
